appsdk/sdk.go

```diff
@@ -142,6 +214,22 @@ func (sdk *AppFunctionsSDK) Initialize() error {
 
 	flag.Parse()
 
+	// Service keys must be unique. If an executable is run multiple times, it must have a different
+	// profile for each instance, thus adding the profile to the base key will make it unique.
+	// This requires services that are expected to have multiple instances running, such as the Configurable App Service,
+	// add the ProfileSuffixPlaceholder placeholder in the service key.
+	//
+	// The Dockerfile must also take this into account and set the profile appropriately, i.e. not just "docker"
+	//
```

Comment:
I don't think `profile` should be overloaded in this way. The Profile flag has referred to a deployment strategy and here it's being used to denote a given instance. Why not have an `instance` key or something. Otherwise our terminology will get confused. This could also be something the Device Services will need eventually because they too have to be uniquely identified. Right now each device service has its own unique name.

Reply:
@tsconn23 , the context for using it this way is deployment. i.e, deploying the same executable multiple times, each with a different configuration profile. Works great w/o Consul, but when using Consul the service keys need to be unique for each deployed instance. IMHO, the profile name is the natural way to make the service keys unique.

After more thought, this code may better belong in the new app-service-configurable, which is the executable that will be deployed multiple times. 

We plan to have this as an agenda topic from tomorrow's App Services WG meeting.